### ISSUE TITLE
Stats Smiley/Pixel Img Doesn't Have Alt Attribute

### ISSUE THREAD

**Issue by cgrymala** [Automattic/jetpack#2856]
For accessibility purposes (and in order to be valid HTML), all `<img>` tags are required to have an alt attribute, even if that attribute is empty. The stats smiley that gets generated by Jetpack does not seem to have an alt attribute.

The sample smiley HTML currently looks like:

`<img src="http://pixel.wp.com/g.gif?v=ext&amp;j=1%3A3.7.2&amp;blog=19082149&amp;post=769&amp;tz=-4&amp;srv=example.com&amp;host=example.com&amp;ref=&amp;rand=0.1583563662134111" id="wpstats">`

It should look like:

`<img src="http://pixel.wp.com/g.gif?v=ext&amp;j=1%3A3.7.2&amp;blog=19082149&amp;post=769&amp;tz=-4&amp;srv=example.com&amp;host=example.com&amp;ref=&amp;rand=0.1583563662134111" id="wpstats" alt="">`

**Comment by egill** [comment]
@jeherve looks like it's just a change on WordPress.com, e.js and w.js? Want me to jump in and do that or?

**Comment by jeherve** [comment]
> @jeherve looks like it's just a change on WordPress.com, e.js and w.js? Want me to jump in and do that or?

That seems like a safe change, but it might be best to run it by the
Data team to be sure.

@jonathansadowski What do you think?

**Comment by jonathansadowski** [comment]
Thanks @egill, it'd be great if you're able to. I tried looking for any discussion of why an alt tag may have been purposely omitted, but couldn't find one, and I can't think of any reason we wouldn't want to include a tag, so :+1: from me.

**Comment by jeherve** [comment]
@egill It might be worth addressing #1963 as well at the same time, since the 2 changes are related.

**Comment by jeherve** [comment]
Also reported here:
https://wordpress.org/support/topic/ggif-fails-accessibility-test?replies=1&view=all

**Comment by jeherve** [comment]
Fixed in r134484-wpcom

**Comment by kraftbj** [comment]
Received a report that the alt text should not have ASCII art, including smileys passed to us via p1637291268091300-slack-CDD9LQRSN

We'd need to fix `e.js` directly, but `w.js` needs to be built/minified via https://github.com/Automattic/analytics/search?q=alt before going to wp.com.

**Comment by terriann** [comment]
The WAVE accessibility evaluation tool (https://wave.webaim.org/) identified this as an error.

Here's the output that it provides when this image with the alt tag `alt=":)"` is found:

> **Errors**
> Missing alternative text
> **What It Means**
> Image alternative text is not present.
> **Why It Matters**
> Each image must have an alt attribute. Without alternative text, the content of an image will not be available to screen reader users or when the image is unavailable.
> **What To Do**
> Add an alt attribute to the image. The attribute value should accurately and succinctly present the content and function of the image. If the content of the image is conveyed in the context or surroundings of the image, or if the image does not convey content or have a function, it should be given empty/null alternative text (alt="").
> **The Algorithm... in English**
> An image does not have an alt attribute.
> **Standards and Guidelines**
> [1.1.1 Non-text Content (Level A)](https://webaim.org/standards/wcag/checklist#sc1.1.1)

**Comment by kraftbj** [comment]
Solved via r235341-wpcom though, due to caching, it may be Monday before a lot of Jetpack sites see the new JS.
